pkg/util/cluster.go

```diff
@@ -56,32 +101,40 @@ func CreateClusterObject(controlPlaneClient *karmadaclientset.Clientset, cluster
 
 // CreateOrUpdateClusterObject create cluster object in karmada control plane,
 // if cluster object has been existed and different from input clusterObj, update it.
-func CreateOrUpdateClusterObject(controlPlaneClient *karmadaclientset.Clientset, clusterObj *clusterv1alpha1.Cluster, mutate func(*clusterv1alpha1.Cluster)) (*clusterv1alpha1.Cluster, error) {
+func CreateOrUpdateClusterObject(controlPlaneClient karmadaclientset.Interface, clusterObj *clusterv1alpha1.Cluster, mutate func(*clusterv1alpha1.Cluster)) (*clusterv1alpha1.Cluster, error) {
 	cluster, exist, err := GetClusterWithKarmadaClient(controlPlaneClient, clusterObj.Name)
 	if err != nil {
 		return nil, err
 	}
-
-	if exist {
-		if reflect.DeepEqual(cluster.Spec, clusterObj.Spec) {
-			klog.Warningf("cluster(%s) already exist and newest", clusterObj.Name)
+	if mutate == nil {
+		klog.Warningf("mutate is nil, report-secrets may be contains KubeCredentials or KubeImpersonator at least, otherwise create cluster object without secret field")
```

Comment:
I didn't get what the log means.
Do you mean we require at least one of `KubeCredentials` or `KubeImpersonator`?

Reply:
> Do you mean we require at least one of KubeCredentials or KubeImpersonator?

i think register a new cluster should require at least one of KubeCredentials or KubeImpersonator, but cluster still can register succcessfully;  Log warning remind user add options, Do you think it's suitable？

Reply:
For `karmada-agent`, it's not necessary to report the secrets if they don't need `aggregated API`(for `KubeImpersonator`) and `karmada-search`(for `KubeCredentials`).

Reply:
> it's not necessary to report the secrets if they don't need aggregated API(for KubeImpersonator) and karmada-search(for KubeCredentials).

so I should remove this log, right?

Reply:
I guess so.